### PR TITLE
feat: Refactor staging to simplify and new deploy (SC-873)

### PR DESCRIPTION
**Pull request by hexonaut**
Do another staging deploy and refactor to eliminate unused code. Unused code is mostly from Sepolia variant.

### CODE REVIEW

File: script/staging/FullStagingDeploy.s.sol [hunk 1]
```diff
@@ -459,50 +440,42 @@ contract StagingDeploymentBase is Script {
...
 
-    function _transferOwnershipOfMocks() internal {
-        vm.selectFork(mainnet.forkId);
-        vm.startBroadcast();
-
-        MockDaiUsds(daiUsds).transferOwnership(mainnetAlmProxy);
-        MockPSM(psm).transferOwnership(mainnetAlmProxy);
-
-        vm.stopBroadcast();
+    function _runFullDeployment() internal {
```
Comment (lucas-manuel) [inline script/staging/FullStagingDeploy.s.sol:443]: This can be removed

File: script/staging/FullStagingDeploy.s.sol [hunk 2]
```diff
@@ -459,50 +440,42 @@ contract StagingDeploymentBase is Script {
...
-        SAFE_BASE = base.config.readAddress(".safe");
-        USDC_BASE = base.config.readAddress(".usdc");
-
-        // Step 3: Run deployment scripts after setting storage variables
-
-        _setUpDependencies(useLiveContracts);
+        _setUpDependencies();
         _setUpAllocationSystem();
         _setUpALMController();
         _setUpBasePSM();
```
Comment (lucas-manuel) [inline script/staging/FullStagingDeploy.s.sol:475]: This can be removed, can use live PSM

File: script/staging/FullStagingDeploy.s.sol [hunk 3]
```diff
@@ -328,6 +291,22 @@ contract StagingDeploymentBase is Script {
...
+            name:       "usdeMintData",
+            decimals:   6
+        });
+        MainnetControllerInit.setRateLimitData({
+            key:        MainnetController(instance.controller).LIMIT_USDE_BURN(),
+            rateLimits: IRateLimits(instance.rateLimits),
+            data:       rateLimitData18,
+            name:       "usdeBurnData",
+            decimals:   18
+        });
```
Comment (lucas-manuel) [inline script/staging/FullStagingDeploy.s.sol:308]: Need to add 4626 deposit/withdraw rate limits for susde

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

File: script/staging/FullStagingDeploy.s.sol (hunk 1 above)
Comment (hexonaut) [reply script/staging/FullStagingDeploy.s.sol:443]: done

File: script/staging/FullStagingDeploy.s.sol (hunk 2 above)
Comment (hexonaut) [reply script/staging/FullStagingDeploy.s.sol:475]: Ah ya good call. Removed a bunch of other mocks in favour of real one.

File: script/staging/FullStagingDeploy.s.sol (hunk 3 above)
Comment (hexonaut) [reply script/staging/FullStagingDeploy.s.sol:308]: ah yes.. ill manually add to current deploy as well

File: script/staging/FullStagingDeploy.s.sol [hunk 4]
```diff
@@ -306,6 +293,26 @@ contract FullStagingDeploy is Script {
             name:       "usdeBurnData",
             decimals:   18
         });
+        MainnetControllerInit.setRateLimitData({
+            key:        RateLimitHelpers.makeAssetKey(
+                            MainnetController(instance.controller).LIMIT_4626_DEPOSIT(),
```
Comment (lucas-manuel) [inline script/staging/FullStagingDeploy.s.sol:298]: Reformatted for easier review:

```solidity
        bytes32 mintKey = MainnetController(instance.controller).LIMIT_USDE_MINT();
        bytes32 burnKey = MainnetController(instance.controller).LIMIT_USDE_BURN();
    
        bytes32 susdsDepositKey = RateLimitHelpers.makeAssetKey(
            MainnetController(instance.controller).LIMIT_4626_DEPOSIT(),
            address(MainnetController(instance.controller).susde())
        );

        bytes32 susdsWithdrawKey = RateLimitHelpers.makeAssetKey(
            MainnetController(instance.controller).LIMIT_4626_WITHDRAW(),
            address(MainnetController(instance.controller).susde())
        );

        // Extra rate limit configuration
        MainnetControllerInit.setRateLimitData(mintKey,     IRateLimits(instance.rateLimits), rateLimitData6,  "usdeMintData",      6);
        MainnetControllerInit.setRateLimitData(burnKey,     IRateLimits(instance.rateLimits), rateLimitData18, "usdeBurnData",      18);
        MainnetControllerInit.setRateLimitData(depositKey,  IRateLimits(instance.rateLimits), rateLimitData18, "susdsDepositData",  18);
        MainnetControllerInit.setRateLimitData(withdrawKey, IRateLimits(instance.rateLimits), ateLimitData18,  "susdsWithdrawData", 18);
```

was able to spot duplicated susdsMintData name after reformatting, updated to deposit/withdraw.

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
These should be output to a different JSON file:

```solidity
        ScriptTools.exportContract(mainnet.name, "allocatorOracle",   oracle);
        ScriptTools.exportContract(mainnet.name, "allocatorRegistry", registry);
        ScriptTools.exportContract(mainnet.name, "allocatorRoles",    roles);

        ScriptTools.exportContract(mainnet.name, "allocatorBuffer", buffer);
        ScriptTools.exportContract(mainnet.name, "allocatorVault",  vault);

        ScriptTools.exportContract(mainnet.name, "dai",      dai);
        ScriptTools.exportContract(mainnet.name, "daiUsds",  daiUsds);
        ScriptTools.exportContract(mainnet.name, "jug",      jug);
        ScriptTools.exportContract(mainnet.name, "psm",      psm);
        ScriptTools.exportContract(mainnet.name, "susds",    susds);
        ScriptTools.exportContract(mainnet.name, "usdc",     usdc);
        ScriptTools.exportContract(mainnet.name, "usds",     usds);
        ScriptTools.exportContract(mainnet.name, "usdsJoin", usdsJoin);
        ScriptTools.exportContract(mainnet.name, "vat",      vat);
```

File: deploy/ControllerInit.sol [hunk 5]
```diff
@@ -296,14 +296,14 @@ library ForeignControllerInit {
...
+        setRateLimitData(RateLimitHelpers.makeAssetKey(withdrawKey, addresses.usdc),  rateLimits, data.usdcWithdrawData,         "usdcWithdrawData",         6);
+        setRateLimitData(RateLimitHelpers.makeAssetKey(depositKey,  addresses.usds),  rateLimits, data.usdsDepositData,          "usdsDepositData",          18);
+        setRateLimitData(RateLimitHelpers.makeAssetKey(withdrawKey, addresses.usds),  rateLimits, data.usdsWithdrawData,         "usdsWithdrawData",         18);
+        setRateLimitData(RateLimitHelpers.makeAssetKey(depositKey,  addresses.susds), rateLimits, data.susdsDepositData,         "susdsDepositData",         18);
+        setRateLimitData(RateLimitHelpers.makeAssetKey(withdrawKey, addresses.susds), rateLimits, data.susdsWithdrawData,        "susdsWithdrawData",        18);
+        setRateLimitData(controller.LIMIT_USDC_TO_CCTP(),        rateLimits, data.usdcToCctpData,           "usdcToCctpData",           6);
+        setRateLimitData(domainKeyEthereum,                      rateLimits, data.cctpToEthereumDomainData, "cctpToEthereumDomainData", 6);
```
Comment (lucas-manuel) [inline deploy/ControllerInit.sol:306]: Out of alignment

Comment (lucas-manuel) [reply deploy/ControllerInit.sol:306]: Can break the last two lines into their own group as well

Comment (hexonaut) [reply deploy/ControllerInit.sol:306]: done

File: deploy/ControllerInit.sol [hunk 6]
```diff
@@ -295,14 +295,15 @@ library ForeignControllerInit {
...
-        _setRateLimitData(domainKeyEthereum,                      rateLimits, data.cctpToEthereumDomainData, "cctpToEthereumDomainData", 6);
+        setRateLimitData(RateLimitHelpers.makeAssetKey(depositKey,  addresses.usdc),  rateLimits, data.usdcDepositData,   "usdcDepositData",          6);
+        setRateLimitData(RateLimitHelpers.makeAssetKey(withdrawKey, addresses.usdc),  rateLimits, data.usdcWithdrawData,  "usdcWithdrawData",         6);
+        setRateLimitData(RateLimitHelpers.makeAssetKey(depositKey,  addresses.usds),  rateLimits, data.usdsDepositData,   "usdsDepositData",          18);
+        setRateLimitData(RateLimitHelpers.makeAssetKey(withdrawKey, addresses.usds),  rateLimits, data.usdsWithdrawData,  "usdsWithdrawData",         18);
+        setRateLimitData(RateLimitHelpers.makeAssetKey(depositKey,  addresses.susds), rateLimits, data.susdsDepositData,  "susdsDepositData",         18);
+        setRateLimitData(RateLimitHelpers.makeAssetKey(withdrawKey, addresses.susds), rateLimits, data.susdsWithdrawData, "susdsWithdrawData",        18);
+
```
Comment (lucas-manuel) [inline deploy/ControllerInit.sol:304]:
```suggestion
        setRateLimitData(RateLimitHelpers.makeAssetKey(depositKey,  addresses.usdc),  rateLimits, data.usdcDepositData,   "usdcDepositData",   6);
        setRateLimitData(RateLimitHelpers.makeAssetKey(withdrawKey, addresses.usdc),  rateLimits, data.usdcWithdrawData,  "usdcWithdrawData",  6);
        setRateLimitData(RateLimitHelpers.makeAssetKey(depositKey,  addresses.usds),  rateLimits, data.usdsDepositData,   "usdsDepositData",   18);
        setRateLimitData(RateLimitHelpers.makeAssetKey(withdrawKey, addresses.usds),  rateLimits, data.usdsWithdrawData,  "usdsWithdrawData",  18);
        setRateLimitData(RateLimitHelpers.makeAssetKey(depositKey,  addresses.susds), rateLimits, data.susdsDepositData,  "susdsDepositData",  18);
        setRateLimitData(RateLimitHelpers.makeAssetKey(withdrawKey, addresses.susds), rateLimits, data.susdsWithdrawData, "susdsWithdrawData", 18);
```

Comment (hexonaut) [reply deploy/ControllerInit.sol:304]: fixed

**Pull request merged**